src/app/templates/TransactionTabs/use-evm-estimation-form.ts

```diff
@@ -0,0 +1,220 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { isDefined } from '@rnw-community/shared';
+import { transform } from 'lodash';
+import { useForm } from 'react-hook-form-v7';
+import { useDebounce } from 'use-debounce';
+import {
+  FeeValuesEIP1559,
+  FeeValuesLegacy,
+  TransactionSerializable,
+  formatEther,
+  parseEther,
+  serializeTransaction
+} from 'viem';
+
+import { SEND_ETH_GAS_LIMIT } from 'lib/constants';
+import { getEvmBalancesChanges } from 'lib/evm/on-chain/get-evm-balances-changes';
+import { useTypedSWR } from 'lib/swr';
+import { EvmEstimationDataWithFallback, StoredAccount } from 'lib/temple/types';
+import { AccountForChain, getAccountAddressForEvm } from 'temple/accounts';
+import { getReadOnlyEvmForNetwork } from 'temple/evm';
+import { isEvmEstimationData } from 'temple/evm/utils';
+import { useAllEvmChains } from 'temple/front';
+import { BalancesChanges, TempleChainKind } from 'temple/types';
+
+import { DEFAULT_INPUT_DEBOUNCE } from './constants';
+import { useEvmEstimationDataState } from './context';
+import { useEvmFeeOptions } from './hooks/use-evm-fee-options';
+import { EvmTxParamsFormData, FeeOptionLabel, Tab } from './types';
+
+const serializeBigint = (value: bigint | nullish) => (typeof value === 'bigint' ? value.toString() : undefined);
+
+export const useEvmEstimationForm = (
+  estimationData: EvmEstimationDataWithFallback | undefined,
+  basicParams: TransactionSerializable | undefined,
+  senderAccount: StoredAccount | AccountForChain<TempleChainKind.EVM>,
+  chainId: number,
+  simulateOperation?: boolean
+) => {
+  const fullEstimationData = isEvmEstimationData(estimationData) ? estimationData : undefined;
+  const accountAddress = useMemo(
+    () => ('address' in senderAccount ? (senderAccount.address as HexString) : getAccountAddressForEvm(senderAccount)!),
+    [senderAccount]
+  );
+  const chains = useAllEvmChains();
+  const chain = chains[chainId];
```

Comment:
mb it would be better to use `.at()` here? is it possible for dApp to send confirmation request to the wallet with chainId that is not approved / present in it?